test/jdk/sun/security/util/HexDumpEncoderTests.java

```diff
@@ -0,0 +1,124 @@
+import sun.security.util.HexDumpEncoder;
+
+import java.io.FileInputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+
+/*
+ * @test
+ * @bug 8349664
+ * @summary HEX dump should always use ASCII or ISO_8859_1
+ * @modules java.base/sun.security.util
+ * @library /test/lib
+ */
+public class HexDumpEncoderTests {
+
+    private final static String CERT_PATH = Path.of(System.getProperty("test.src", "."))
+            .resolve("HostnameChecker")
+            .resolve("cert5.crt")
+            .toString();
+
+    private static String[] getTestCommand(final String encoding) {
+        return new String[]{
+                "--add-modules", "java.base",
+                "--add-exports", "java.base/sun.security.util=ALL-UNNAMED",
+                "-Dfile.encoding=" + encoding,
+                HexDumpEncoderTests.HexDumpEncoderTest.class.getName(),
+                CERT_PATH};
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        final var testCommandIso = getTestCommand("ISO-8859-1");
+
+        final var resultIso = ProcessTools.executeTestJava(testCommandIso);
+        resultIso.shouldHaveExitValue(0);
+
+        // This will take all available StandardCharsets and test them all comparing to the ISO_8859_1
+        // Dome im parallel, as this is significantly faster
+        Arrays.stream(StandardCharsets.class.getDeclaredFields())
+                .parallel()
+                .forEach(field -> {
+                    if (java.lang.reflect.Modifier.isStatic(field.getModifiers())) {
+                        try {
+                            final var charset = (Charset) field.get(StandardCharsets.ISO_8859_1); // getting the charset to test
+
+                            final var testCommand = getTestCommand(charset.name());
+
+                            final var result = ProcessTools.executeTestJava(testCommand);
+                            result.shouldHaveExitValue(0);
+
+                            // The outputs of the ISO encoding must be identical to the one tested
+                            Asserts.assertEquals(resultIso.getStdout(),
+                                    result.getStdout(),
+                                    "Encoding " + charset.name());
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                });
+    }
+
+    public static class HexDumpEncoderTest {
+
+        /**
+         * This will test the encode and encode buffer functions at once,
+         * as they are both representing the string in LATIN_1
+         * <p>
+         * The output is put as a system.out
+         */
+        public static void main(String[] args) throws Exception {
+
+            try (final FileInputStream fis = new FileInputStream(args[0])) {
```

Comment:
There is no need to read from a certificate. I think `encoder.encodeBuffer(new byte[100])` is enough.

Reply:
Some `import`s are not needed.